Verify that all nodes have started in ClusterBase.Start

MaxKsyunz opened the pull request:
ClusterBase.Start was failing to detect when all nodes have started. 
The root caused turned out to be in OpenSearch log parsing. 
It needed to be updated to match what OpenSearch produced. 
In addition, it did not support nodes listening on both IPv4 and IPv6 addresses.

### Additional Reviewers
@guiangumpac 
@Yury-Fridlyand 
@alexmeizer 
@raymond-lum

Commit:
Wait for cluster nodes to start.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Yury-Fridlyand committed:
Update readme for xunit - remove elastic reference

Signed-off-by: Yury Fridlyand <yuryf@bitquilltech.com>

Commit:
Update log parsing to match OpenSearch

Update log parsing in OpenSearchNode to look for format that OpenSearch
produces.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

guiangumpac reviewed (changes requested):
Works well with OpenSearch but there needs to be a check for serverType because opendistro has the old log patterns.

alexmeizer committed:
Merge pull request #3 from Bit-Quill/update-xunit-readme

Update readme for xunit - remove elastic reference

Commit:
Changed security config and running server check

Signed-off-by: Guian Gumpac <guiang@bitquilltech.com>

Commit:
Reverted change for enableSSL check

Signed-off-by: Guian Gumpac <guiang@bitquilltech.com>

Commit:
Add ServerType property to ClusterConfiguration

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Convert LineOutParser to non-static class

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Use LineOutParser specific to server type

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Update xUnit to 2.4.1

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Fix code style

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Return non-successful HttpResponse

If HTTP request in  ClusterCompositeTask.Call gets
an unseccessul status code, return it for the client
to handle.

In a simiar vein, do not swollow exceptions and let
the client code handle them.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Retry failed requests in ValidateRunningVersion

Update ValidateRunningVersion.Run to retry Http requests
if they fail.

Currently, the method will retry 4 times if the initial request
did not get a successful response.

It will wait 5+ seconds between each retry.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Server-specific instance of LineOutParser

Use different version of LineOutParser depending on
server type.

OpenSearch and OpenDistro produce slightly different log output.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Add integration tests for EphemeralCluster

Add an integration test that verifies that EphemeralCluster.Start
correctly picks up when an ephemeral cluster is started.

The test checks a version of OpenDistroForElastic and OpenSearch
with security on and off.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Changing newlines for license check to pass.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>

Commit:
Updated license header to match expected text

There was a Unicode byte order mark in the cs file.
It caused license header check to fail.

Signed-off-by: MaxKsyunz <maxk@bitquilltech.com>
